cgi-bin/sitemap.js

```diff
@@ -10,12 +10,61 @@
  * governing permissions and limitations under the License.
  */
 const algoliasearch = require('algoliasearch');
+const pick = require('lodash.pick');
+const rp = require('request-promise-native');
 const { logger } = require('@adobe/openwhisk-action-logger');
 const { wrap } = require('@adobe/openwhisk-action-utils');
 const Downloader = require('@adobe/helix-pipeline/src/utils/Downloader.js');
 const { MountConfig, IndexConfig } = require('@adobe/helix-shared');
 const { getOriginalHost } = require('../src/utils');
 
+/**
+ * Create the search provider that will return all paths for the sitemap.
+ *
+ * @param {object} index index configuration
+ * @param {object} params parameters relevant for the provider
+ * @returns provider instance
+ */
+function createSearchProvider(index, params) {
+  const { owner, repo } = params;
+
+  const { sitemap } = index;
+  if (sitemap) {
+    return {
+      search: async (query, {
+        hitsPerPage,
+        page,
+        attributesToRetrieve,
+      }) => {
+        const uri = index.fetch.replace(/\{owner\}/g, owner).replace(/\{repo\}/g, repo).replace(/\{path\}/g, sitemap);
+        const qs = {
+          limit: hitsPerPage,
+          offset: page * hitsPerPage,
```

Comment:
does page start a 0 ?

Reply:
No, it starts at 1: https://github.com/adobe/helix-pages/blob/master/htdocs/sitemap.xml

Reply:
so then this is wrong